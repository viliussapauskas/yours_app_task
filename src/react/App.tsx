import React from "react";
import "./App.scss";
import { HomePage } from "../react/homePage";
import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
import { IssuesPage } from "./issuesPage";

const App = () => {
  return (
    <Router>
      <div className="App">
        <Routes>
          <Route path="/" element={<HomePage />} />
          <Route path="/issues" element={<IssuesPage />} />
        </Routes>
      </div>
    </Router>
  );
};

export default App;
