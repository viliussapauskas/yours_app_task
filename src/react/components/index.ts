export { Layout } from './layout'
export { Button } from './button'
export { TextInput } from './textInput'
