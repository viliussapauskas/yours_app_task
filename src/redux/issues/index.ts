export { fetchIssues } from './issuesAPI'
export {
    issuesReducer,
    isssuesActionCreators,
    issuesSlice,
    getIssuesStateSelector,
    loadIssuesAsync,
} from './issuesSlice'
export { initialState } from './state'
export { type Issue, type IssuesPayload, type IssuesState } from './types'
