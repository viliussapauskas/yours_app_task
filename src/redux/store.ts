import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import { issuesReducer } from "../redux/issues";

export const store = configureStore({
  reducer: {
    issues: issuesReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
