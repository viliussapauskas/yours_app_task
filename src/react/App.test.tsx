import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import App from './App'
import { store } from '../redux/store'

const renderComponent = () => {
    return render(
        <Provider store={store}>
            <App />
        </Provider>
    )
}

describe('App tests', () => {
    it('Should render component', () => {
        expect(renderComponent()).toMatchSnapshot()
    })
})
