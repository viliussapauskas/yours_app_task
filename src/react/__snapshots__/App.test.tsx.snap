// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests Should render component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div
          class="layout layout"
          data-testid="layout-container"
        >
          <div
            class="logo"
          >
            <svg>
              logo.svg
            </svg>
          </div>
          <div
            class="child"
            data-testid="layout-child-container"
          >
            <div
              class="container"
              data-testid="home-page"
            >
              <div
                class="inputsContainer"
              >
                <div
                  class="container"
                  data-testid="text-input-container"
                >
                  <label
                    class="label"
                  >
                    Owner
                    <span
                      style="color: red;"
                    >
                       *
                    </span>
                  </label>
                  <input
                    class="textInput"
                    data-testid="text-input"
                    id="Owner"
                    placeholder="Owner name"
                    type="text"
                  />
                </div>
                <div
                  class="container"
                  data-testid="text-input-container"
                >
                  <label
                    class="label"
                  >
                    Repository
                    <span
                      style="color: red;"
                    >
                       *
                    </span>
                  </label>
                  <input
                    class="textInput"
                    data-testid="text-input"
                    id="Repository"
                    placeholder="Repository name"
                    type="text"
                  />
                </div>
              </div>
              <div>
                <button
                  class="button"
                  data-testid="button-component"
                  type="button"
                >
                  Show issues
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div
        class="layout layout"
        data-testid="layout-container"
      >
        <div
          class="logo"
        >
          <svg>
            logo.svg
          </svg>
        </div>
        <div
          class="child"
          data-testid="layout-child-container"
        >
          <div
            class="container"
            data-testid="home-page"
          >
            <div
              class="inputsContainer"
            >
              <div
                class="container"
                data-testid="text-input-container"
              >
                <label
                  class="label"
                >
                  Owner
                  <span
                    style="color: red;"
                  >
                     *
                  </span>
                </label>
                <input
                  class="textInput"
                  data-testid="text-input"
                  id="Owner"
                  placeholder="Owner name"
                  type="text"
                />
              </div>
              <div
                class="container"
                data-testid="text-input-container"
              >
                <label
                  class="label"
                >
                  Repository
                  <span
                    style="color: red;"
                  >
                     *
                  </span>
                </label>
                <input
                  class="textInput"
                  data-testid="text-input"
                  id="Repository"
                  placeholder="Repository name"
                  type="text"
                />
              </div>
            </div>
            <div>
              <button
                class="button"
                data-testid="button-component"
                type="button"
              >
                Show issues
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
